lib/intercom/request.rb

```diff
@@ -152,6 +152,8 @@ def raise_application_errors_on_failure(error_list_details, http_code)
       case error_code
       when 'unauthorized', 'forbidden', 'token_not_found'
         raise Intercom::AuthenticationError.new(error_details['message'], error_context)
+      when 'token_suspended'
```

Comment:
Do you want add a check for `token_unauthorized` and `token_revoked` too?

Reply:
Ah good shout, my bad!

Reply:
In this case, I think it's better to only check for `token_revoked` as `token_unauthorized` is already handled with the core AuthenticationError class